linode/manager
# upcoming: [M3-8378] - Add Bucket Management `Properties` Tab for Object Storage Gen2

## Description 📝
We aim to enhance our bucket management interface by introducing a new "Properties" tab. This tab will display the current rate limits for each bucket and include functionality to modify these limits in future updates.

## Changes  🔄
- Add Bucket Management Properties Tab for Object Storage Gen2.

## Target release date 🗓️
N/A

## Preview 📷
![Screenshot 2024-08-22 at 1 05 48 AM](https://github.com/user-attachments/assets/4c256444-5e3f-4e6d-b7c9-f82cda10c1c0)

## How to test 🧪

### Prerequisites
- Enable MSW and ensure OBJ Gen2 flag is enabled.

### Verification steps
> [!Note]
> - The endpoints for retrieving and updating Bucket Rate Limits are currently unavailable.
> - Hardcoded values are being used to retrieve Bucket Rate Limits data.

- Navigate to http://localhost:3000/object-storage/buckets
- Select a bucket.
- When on details page, verify that the new 'Properties' tab is rendered.
- Verify that the user interface correctly displays the retrieved Bucket Rate Limits data.
- Verify that the functionality to update Bucket Rate Limits has been implemented and is ready for integration with the API endpoints when they become available.

## As an Author I have considered 🤔

*Check all that apply*
